src/hardware/pros_motor_group.cpp

```diff
@@ -0,0 +1,47 @@
+#include "hardware/pros_motor_group.h"
+
+namespace hardware {
+void ProsMotorGroup::Move(int value) { motors().move(value); }
+void ProsMotorGroup::MoveVoltage(int voltage) {
+  motors().move_voltage(voltage);
+}
+void ProsMotorGroup::MoveVelocity(float velocity) {
+  float rpm = (velocity / constant::kMaxVelocity) * max_rpm();
+  motors().move_velocity(rpm);
+}
+void ProsMotorGroup::MoveAbsolute(double position, int max_velocity) {
+  motors().move_absolute(position, max_velocity);
+}
+void ProsMotorGroup::ResetEncoder() { motors().tare_position(); }
+
+double ProsMotorGroup::position() const {
+  // manual member access to avoid automatically calling const motors()
+  std::vector<double> positions = motors_.get_positions();
+  return std::reduce(positions.begin(), positions.end()) / positions.size();
+}
+float ProsMotorGroup::velocity() const {
+  std::vector<double> motor_rpms = motors_.get_actual_velocities();
+  float rpm =
+      std::reduce(motor_rpms.cbegin(), motor_rpms.cend()) / motor_rpms.size();
+  return (rpm / max_rpm()) * constant::kMaxVelocity;
+}
+
+std::vector<std::int8_t> ProsMotorGroup::FlipPortNumbers(
+    std::vector<int> port_numbers, std::vector<bool> reverse) {
```

Comment:
thoughts on using an array of structs instead of two separate vectors?

Reply:
Yeah, I think a struct makes a lot of sense. I was debating reserving the struct for the factory, but that doesn't really make sense if we can't do dependency injection.